src/java.base/share/classes/java/io/Reader.java

```diff
@@ -184,12 +184,34 @@ protected Reader(Object lock) {
      * @since 1.5
      */
     public int read(CharBuffer target) throws IOException {
-        int len = target.remaining();
-        char[] cbuf = new char[len];
-        int n = read(cbuf, 0, len);
-        if (n > 0)
-            target.put(cbuf, 0, n);
-        return n;
+        int nread;
+        if (target.hasArray()) {
+            char cbuf[] = target.array();
+            int off = target.arrayOffset() + target.position();
+            int len = target.remaining();
+            nread = this.read(cbuf, off, len);
+            if (nread > 0) {
+                target.position(target.position() + nread);
+            }
+        } else {
+            int remaining = target.remaining();
+            char cbuf[] = new char[Math.min(remaining, TRANSFER_BUFFER_SIZE)];
```

Comment:
Would there be value in making this a (lazily created) member variable? That would allow a single instance to be reused. It seems likely that, if one call is made with a direct CharBuffer, subsequent calls will also be made with direct instances (probably same instance?).

Reply:
I am not sure. It would help to reduce the allocation rate when reading a large amount of data into a small direct CharBuffer. I don't know how common that is. We would introduce an instance variable for one path in one method.